### PR TITLE
File rename

### PR DESCRIPTION
**Pull request by CiprianRegus**
- Moved header files from include to include/no_os directory.
- Updated header and source files with the correct include paths.
- Updated the include paths in src.mk files.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
nit: some commit descriptions would be nice :)
Also please sign the CLA since you're not yet part of the org.

The CI detected issues seem to be unrelated to the changes, so it should be fine like this.

**Comment by adrimbarean** [comment]
nitpick: looks like ad9361_conv.c and ad9361_util.c could be AStyled. I'd do this in a separate commit, but since it's not originally part of this pull request we can also skip it. Choice is yours.

**Review by adrimbarean — APPROVED** [review]
Looks good to me. Just a comment.

File: drivers/platform/linux/linux_uart.c [hunk 1]
```diff
@@ -209,7 +209,7 @@ int32_t uart_init(struct uart_desc **desc, struct uart_init_param *param)
 		goto free;
 	}
 
-	if (param->stop == UART_STOP_1)
+	if (param->stop == UART_STOP_1BIT)
```
Comment (mchindri) [inline drivers/platform/linux/linux_uart.c:212]: Add in the commit description the reason of the change too

Comment (mchindri) [reply drivers/platform/linux/linux_uart.c:212]: Same for previous commit

**Review by amiclaus — APPROVED** [review]

**Comment by CiprianRegus** [comment]
V3:
- AStyled ad9361_conv.c and ad9361_util.c.

**Review by buha — CHANGES_REQUESTED** [review]
This breaks the project_build target. A little bit of tweaking is needed for those generic.mk changes to make this work with the builds from SDK.

**Comment by buha** [comment]
Screenshot from 2021-12-15 09-40-41
Screenshot from 2021-12-15 09-42-06

**Comment by dbogdan** [comment]
I would use the same name as the one of the repository: "no-os" instead of "no_os".

**Review by amiclaus — COMMENTED** [review]
@mchindri is it possible to add your commits in a separate PR?

**Comment by buha** [comment]
I tested projects running on xilinx and stm32 and the include problem in SDK was fixed by your changes

remaining changes:
- squash the commits by Mihail into your big commit that updates generic.mk
- edit ad5758 project and driver which were merged meanwhile to master and don't contain no-os/....h path

**Comment by CiprianRegus** [comment]
V4:
- Edited drivers/dac/ad5758 files to include the correct paths.
- Squashed commits that modify generic.mk.
- Changed the commit title to reflect the changed files.

**Comment by CiprianRegus** [comment]
V5:
- Solved conflicts with recent master commits.
- Moved include/no-os/uart.h file to the correct commit.

File: include/no-os/uart.h [hunk 2]
```diff
@@ -0,0 +1,162 @@
+/***************************************************************************//**
```
Comment (buha) [inline include/no-os/uart.h:1]: this file seems to have been deleted in a previous commit, only to readded it here, this is not ok

**Comment by CiprianRegus** [comment]
V6:
- Fixed include paths in drivers/platform/stm32.
- Solved the issue with the deleted uart.h file in a previous commit.

**Review by buha — APPROVED** [review]

**Pull request merged**